back/tests/cmd/openbsrv-integration/integration_test.go

```diff
@@ -0,0 +1,426 @@
+		tests := []struct {
+			addReq *pb.AddPostReq
+			want   *pb.PostResp
+			fndReq *pb.FndPostsReq
+		}{
+			{
+				&pb.AddPostReq{
+					Title:  "test title postA first",
+					Body:   "test body of first post",
+					TypeId: "2",
+				},
+				&pb.PostResp{
+					Title:  "test title postA first",
+					Body:   "test body of first post",
+					TypeId: "2",
+				},
+				&pb.FndPostsReq{Keywords: []string{
+					"postA",
+					"first",
+					"multiple",
+					"keywords",
+					"not",
+					"available",
+					"yet",
+				},
+				},
+			},
+			{
+				&pb.AddPostReq{
+					Title:  "test title postB second",
+					Body:   "test body of second postB",
+					TypeId: "3",
+				},
+				&pb.PostResp{
+					Title:  "test title postB second",
+					Body:   "test body of second postB",
+					TypeId: "3",
+				},
+				&pb.FndPostsReq{Keywords: []string{"postB"}},
+			},
+		}
+
+		for _, tc := range tests {
+			r, err := clnt.AddPost(ctx, tc.addReq)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			got := r
+			wantPostID := got.Id
+
+			// Unset fields that aren't being tested.
+			got.Id = ""
+			got.Slug = ""
+			got.Created = &timestamp.Timestamp{}
+			got.Updated = &timestamp.Timestamp{}
+			got.Deleted = &timestamp.Timestamp{}
+			got.Blocked = &timestamp.Timestamp{}
+			got.XXX_unrecognized = []byte{}
+			got.XXX_sizecache = 0
+
+			if reflect.DeepEqual(got, tc.want) {
+				t.Errorf("got: %v, want: %v", got, tc.want)
+			}
+
+			gotPostID, err := postSvcFndPost(ctx, conn, clnt, tc.fndReq)
+			if err != nil {
+				t.Error(err)
+			}
+
+			if gotPostID != wantPostID {
+				t.Fatalf("got: %s, want: %s", gotPostID, wantPostID)
+			}
+		}
+	}
+}
+
+func postSvcFndPost(ctx context.Context, conn *grpc.ClientConn, clnt pb.PostClient, req *pb.FndPostsReq) (string, error) {
+	postID, err := clnt.FndPosts(ctx, req)
+	if err != nil {
+		return "", fmt.Errorf("Unable to find post: %w", err)
+	}
+
+	if len(postID.Posts) > 0 {
+		return postID.Posts[0].Id, nil
+	}
+	return "", nil
+}
+
+func postSvcEdtPostFn(ctx context.Context, conn *grpc.ClientConn, clnt pb.PostClient) func(*testing.T) {
+	return func(t *testing.T) {
+		addReq := &pb.AddPostReq{
+			Title:  "Post D",
+			Body:   "This is post D.",
+			TypeId: "2",
+		}
+		_, err := clnt.AddPost(ctx, addReq)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		fndReq := &pb.FndPostsReq{Keywords: []string{"post D"}}
+		postID, err := postSvcFndPost(ctx, conn, clnt, fndReq)
+		if err != nil {
+			t.Error(err)
+		}
+
+		editReq := &pb.AddPostReq{
+			Title:  "Post D (edited)",
+			Body:   "This is post D after edits.",
+			TypeId: "2",
+		}
+
+		ovrReq := &pb.OvrPostReq{Id: postID, Req: editReq}
+		r, err := clnt.OvrPost(ctx, ovrReq)
+		if err != nil {
+			t.Error(err)
+		}
+
+		want := &pb.PostResp{
+			Title:  editReq.Title,
+			Body:   editReq.Body,
+			TypeId: editReq.TypeId,
+		}
+
+		got := r
+
+		// Unset fields that aren't being tested.
+		got.Id = ""
+		got.Slug = ""
+		got.Created = &timestamp.Timestamp{}
```

Comment:
It's curious that these are passing. They may not be. Please check with this play for consideration: https://play.golang.org/p/m9NpE5IBCIE

Reply:
I was able to replace the `&timestamp.Timestamp{}` with `nil` and the tests passed.

However, I could not set the `[]byte` to nil. The tests did not pass with that.